src/Components/Components/src/ComponentFactory.cs

```diff
@@ -25,7 +26,12 @@ internal class ComponentFactory
 
         public IComponent InstantiateComponent(IServiceProvider serviceProvider, Type componentType)
         {
-            var instance = Activator.CreateInstance(componentType);
+            var activator = serviceProvider.GetService<IComponentActivator>();
+
+            var instance = activator != null
+                ? activator.CreateInstance(componentType)
+                : Activator.CreateInstance(componentType);
```

Comment:
Instead of asking the service provider for `IComponentActivator` on every call, we'll need a way of caching the activator instance. This might force us to eliminate the static `ComponentFactory.Instance` and instead have the renderer get its own separate component factory instance during its own construction.

If you're able to look into doing that then great, otherwise I'll see what I can do to fix this up.

Reply:
It would be great if you would look into this after I merge the changes. You have a lot more insight of what needs to be changed and where.